simapp/app.go

```diff
@@ -432,7 +432,7 @@ func NewSimApp(
 // simapp. It is useful for tests and clients who do not want to construct the
 // full simapp
 func MakeCodecs() (codec.Marshaler, *codec.LegacyAmino) {
-	config := MakeEncodingConfig()
+	config := MakeTestEncodingConfig()
```

Comment:
```suggestion
	config := simappparams.MakeEncodingConfig()
	std.RegisterLegacyAminoCodec(config.Amino)
	std.RegisterInterfaces(config.InterfaceRegistry)
	ModuleBasics.RegisterLegacyAminoCodec(config.Amino)
	ModuleBasics.RegisterInterfaces(config.InterfaceRegistry)
```

How about:
1. Move `simapp/encoding.go` to `testutil/encoding.go` (That's where it really belongs)
2. Make the above change.

Advantages:
- `MakeTestEncodingConfig` lives in `testutil`, which makes sense. Some tests which use `MakeTestEncodingConfig` don't even need to import simapp.
- For users looking at simapp as a reference implementation, they will do registration directly in `NewSimApp` as above. This imports `MakeEncodingConfig` from `./params`, which is okay.

Disadvantages:
- We write twice the registration (once for tests, once for reference). It's only executed once though).

Reply:
I like this approach! :+1:

Reply:
This is what we want to do in #7642. This PR, as noted in the description, is a temporal solution for that.

Reply:
In fact what we want to do in #7642 is to hide the `MakeTestEncodingConfig` in a private method, and use `app.AppCodecs` outside.